src-ui.v3/src/PkgId.hs

```diff
@@ -145,3 +146,6 @@ verToText (Ver x) = T.pack . Ver.showVersion . Ver.makeVersion $ x
 ----------------------------------------------------------------------------
 newtype TagN = TagN { tagNToText :: Text }
   deriving (Eq,Ord,FromJSON,ToJSON,ToJSONKey,FromJSONKey,FromHttpApiData,ToHttpApiData)
+
+unTagN :: TagN -> Text
```

Comment:
Why do you need this when you have the same function as tagNToText on line 147?

Reply:
At first I put it under where statement, I think I follow the `unPkgN` and don't realized that pkgNtoText is also already there. Then I move the `unTagN` from the `where` to this module. Does it have any different?

Reply:
You could just remove unTagN and use tagNToText instead. Or rename the record accessor to unTagN if that makes more sense.

Reply:
> You could just remove unTagN and use tagNToText instead. Or rename the record accessor to unTagN if that makes more sense.
Yes thanks ben, I already cleaning some of part in my new PR below.